### PR TITLE
add dy2static support of LayoutLMv2 SER and LayoutXLM SER

### PR DESCRIPTION
**Pull request by WenmuZhou**
### PR types
New features 
### PR changes
Models

### Description
add dy2static support of LayoutLMv2 and LayoutxLM

Use the following commands to complete the dynamic and static transformation of the model
- LayoutXLM
```python
import paddlenlp as ppnlp
model = ppnlp.transformers.LayoutXLMForTokenClassification.from_pretrained("layoutxlm-base-uncased")
model.eval()
model = paddle.jit.to_static(
        model,
        input_spec=[
            paddle.static.InputSpec(
                shape=[None, 512], dtype="int64"),  # input_ids
            paddle.static.InputSpec(
                shape=[None, 512, 4], dtype="int64"),  # bbox
            paddle.static.InputSpec(
                shape=[None, 3, 224, 224], dtype="int64"),  # image
            paddle.static.InputSpec(
                shape=[None, 512], dtype="int64"),  # attention_mask
        ])

# Save in static graph model.
save_path = os.path.join(args.output_path, "inference")
paddle.jit.save(model, 'layoutxlm')
```
- LayoutLMv2
```python
import paddlenlp as ppnlp
model = ppnlp.transformers.LayoutXLMForTokenClassification.from_pretrained("layoutlmv2-base-uncased")
model.eval()
model = paddle.jit.to_static(
        model,
        input_spec=[
            paddle.static.InputSpec(
                shape=[None, 512], dtype="int64"),  # input_ids
            paddle.static.InputSpec(
                shape=[None, 512, 4], dtype="int64"),  # bbox
            paddle.static.InputSpec(
                shape=[None, 3, 224, 224], dtype="int64"),  # image
            paddle.static.InputSpec(
                shape=[None, 512], dtype="int64"),  # attention_mask
        ])

# Save in static graph model.
save_path = os.path.join(args.output_path, "inference")
paddle.jit.save(model,'layoutlmv2')
```

### CODE REVIEW

File: paddlenlp/transformers/layoutlmv2/modeling.py [hunk 1]
```diff
@@ -339,7 +339,7 @@ def __init__(self, config):
         self.dropout = nn.Dropout(config["attention_probs_dropout_prob"])
 
     def transpose_for_scores(self, x):
-        new_x_shape = x.shape[:-1] + [
```
Comment (littletomatodonkey) [inline paddlenlp/transformers/layoutlmv2/modeling.py:342]: 直接用paddle.shape是否就ok，避免list

File: paddlenlp/transformers/layoutlmv2/modeling.py [hunk 2]
```diff
@@ -389,23 +389,30 @@ def forward(
...
             attention_scores)
+
         attention_probs = F.softmax(attention_scores, axis=-1)
         # This is actually dropping out entire tokens to attend to, which might
         # seem a bit unusual, but is taken from the original Transformer paper.
         attention_probs = self.dropout(attention_probs)
         context_layer = paddle.matmul(attention_probs, value_layer)
         context_layer = context_layer.transpose([0, 2, 1, 3])
-        new_context_layer_shape = context_layer.shape[:-2] + [
+        new_context_layer_shape = list(context_layer.shape[:-2]) + [
```
Comment (littletomatodonkey) [inline paddlenlp/transformers/layoutlmv2/modeling.py]: 同上，paddle.shape直接试下？

File: paddlenlp/transformers/layoutxlm/modeling.py [hunk 3]
```diff
@@ -335,9 +335,10 @@ def __init__(self, config):
         self.dropout = nn.Dropout(config["attention_probs_dropout_prob"])
 
     def transpose_for_scores(self, x):
-        new_x_shape = x.shape[:-1] + [
+        new_x_shape = list(x.shape[:-1]) + [
```
Comment (littletomatodonkey) [inline paddlenlp/transformers/layoutxlm/modeling.py:338]: 同上

File: paddlenlp/transformers/layoutxlm/modeling.py [hunk 4]
```diff
@@ -385,23 +386,28 @@ def forward(
...
+            paddle.ones(attention_scores_shape) * float("-1e10"),
             attention_scores)
         attention_probs = F.softmax(attention_scores, axis=-1)
         # This is actually dropping out entire tokens to attend to, which might
         # seem a bit unusual, but is taken from the original Transformer paper.
         attention_probs = self.dropout(attention_probs)
         context_layer = paddle.matmul(attention_probs, value_layer)
         context_layer = context_layer.transpose([0, 2, 1, 3])
-        new_context_layer_shape = context_layer.shape[:-2] + [
+        new_context_layer_shape = list(context_layer.shape[:-2]) + [
```
Comment (littletomatodonkey) [inline paddlenlp/transformers/layoutxlm/modeling.py]: 同上

File: paddlenlp/transformers/layoutlmv2/modeling.py (hunk 1 above)
Comment (WenmuZhou) [reply paddlenlp/transformers/layoutlmv2/modeling.py:342]: 使用paddle.shape后使用concat扩充维度会报错，暂时只能这么操作

**Review by yingyibiao — APPROVED** [review]
LGTM

**Pull request merged**